### PR TITLE
[ParticleMechanics] Further cleanups

### PR DESCRIPTION
**Pull request by bodhinandach**
Following #5151 as discussed.

### CODE REVIEW

File: applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py [hunk 1]
```diff
@@ -203,20 +200,20 @@ def generate_material_point(self):
...
+        self.material_point_model_part.ProcessInfo = self.grid_model_part.ProcessInfo
+        self.material_point_model_part.SetBufferSize(self.grid_model_part.GetBufferSize())
 
         # Generate MP Element and Condition
-        KratosParticle.GenerateMaterialPointElement(self.grid_model_part, self.initial_material_model_part, self.material_model_part, axis_symmetric_flag, pressure_dofs)
-        KratosParticle.GenerateMaterialPointCondition(self.grid_model_part, self.initial_material_model_part, self.material_model_part)
+        KratosParticle.GenerateMaterialPointElement(self.grid_model_part, self.initial_mesh_model_part, self.material_point_model_part, axis_symmetric_flag, pressure_dofs)
+        KratosParticle.GenerateMaterialPointCondition(self.grid_model_part, self.initial_mesh_model_part, self.material_point_model_part)
 
     def search_element(self):
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py]:
```suggestion
    def __SeachElement(self):
```

File: applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py [hunk 2]
```diff
@@ -236,7 +233,7 @@ def import_constitutive_laws(self):
             materials_imported = False
         return materials_imported
 
-    ### Solver private functions
+    ### Solver protected functions
 
     def _add_model_part_containers(self):
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py]: are those protected or private?

=> check the [Style Guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide#python-style-conventions) for naming of functions

**Review by tteschemacher — DISMISSED** [review]
looks good

**Comment by bodhinandach** [comment]
@philbucher, as discussed

File: applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py [hunk 3]
```diff
@@ -123,148 +122,143 @@ def GetGridModelPart(self):
...
 
-    def get_convergence_criterion(self):
+    def GetConvergenceCriteria(self):
         if not hasattr(self, '_convergence_criterion'):
-            self._convergence_criterion = self._create_convergence_criterion()
+            self._convergence_criterion = self._CreateConvergenceCriteria()
         return self._convergence_criterion
 
-    def get_linear_solver(self):
+    def GetLinearSolver(self):
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py]:
```suggestion
    def _GetLinearSolver(self):
```

File: applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py [hunk 4]
```diff
@@ -123,148 +122,143 @@ def GetGridModelPart(self):
...
         if not hasattr(self, '_linear_solver'):
-            self._linear_solver = self._create_linear_solver()
+            self._linear_solver = self._CreateLinearSolver()
         return self._linear_solver
 
-    def get_builder_and_solver(self):
+    def GetBuilderAndSolver(self):
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py]:
```suggestion
    def _GetBuilderAndSolver(self):
```

File: applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py [hunk 5]
```diff
@@ -123,148 +122,143 @@ def GetGridModelPart(self):
...
         if not hasattr(self, '_builder_and_solver'):
-            self._builder_and_solver = self._create_builder_and_solver()
+            self._builder_and_solver = self._CreateBuilderAndSolver()
         return self._builder_and_solver
 
-    def get_solution_strategy(self):
+    def GetSolutionStrategy(self):
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py]:
```suggestion
    def _GetSolutionStrategy(self):
```

File: applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py [hunk 6]
```diff
@@ -123,148 +122,143 @@ def GetGridModelPart(self):
...
         if not hasattr(self, '_solution_strategy'):
-            self._solution_strategy = self._create_solution_strategy()
+            self._solution_strategy = self._CreateSolutionStrategy()
         return self._solution_strategy
 
-    def generate_material_point(self):
+    def ImportConstitutiveLaws(self):
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py]:
```suggestion
    def __ImportConstitutiveLaws(self):
```

**Review by philbucher — COMMENTED** [review]
only the fcts that are in the python-solver are Public, the rest should be protected or private
otherwise looks good to me

**Review by philbucher — APPROVED** [review]

**Pull request merged**